spec/ffi/async_callback_spec.rb

```diff
@@ -34,4 +34,15 @@ module LibTest
     expect(called).to be true
     expect(v).to eq(0x7fffffff)
   end
+
+  it "sets the name of the thread that runs the callback" do
+    skip "not yet supported on TruffleRuby" if RUBY_ENGINE == "truffleruby"
+    skip "not yet supported on JRuby" if RUBY_ENGINE == "jruby"
+
+    thread_name = nil
+
+    LibTest.testAsyncCallback(proc { thread_name = Thread.current.name }, 0)
```

Comment:
It should be race-free to store `Thread.current` here and call `Thread#name` below. The name is readable even if the thread has terminated.